recipe/meta.yaml

```diff
@@ -67,95 +70,104 @@ outputs:
         - cmake -S . -B ./build -G=Ninja && cmake --build ./build -v
     about:
       home: https://developer.nvidia.com/cuda-toolkit
-      license_file: LICENSE
+      license_file:
+        - {{ PREFIX }}/targets/{{ ctk_targets[target_platform] }}/LICENSE
       license: LicenseRef-NVIDIA-End-User-License-Agreement
       license_url: https://docs.nvidia.com/cuda/eula/index.html
       summary: Compiler for CUDA applications.
       description: |
         Compiler for CUDA applications.
       doc_url: https://docs.nvidia.com/cuda/index.html
+  {% endif %}
 
-  - name: cuda-nvcc_{{ target_platform }}
+  - name: cuda-nvcc_{{ cross_target_platform }}
     build:
-      noarch: generic
-    run_exports:
-      strong:
-        - cuda-version >={{ cuda_version }},<{{ cuda_version.split(".")[0]|int + 1 }}
+      run_exports:
+        strong:
+          - cuda-version >={{ cuda_version }},<{{ cuda_version.split(".")[0]|int + 1 }}
     files:
       - etc/conda/activate.d/~cuda-nvcc_activate.sh      # [linux]
       - etc/conda/deactivate.d/~cuda-nvcc_deactivate.sh  # [linux]
     requirements:
       run:
+        # Safeguard for early parsing
+        {% if cross_target_platform is defined %}
+        - {{ pin_subpackage("cuda-nvcc-impl_" + cross_target_platform, exact=True) }}
```

Comment:
From what I can tell, this isn't a legal dependency? The package `"cuda-nvcc-impl_" + cross_target_platform` is only defined when `target_platform == cross_target_platform`.

I think we may need to define _all_ the cross-packages for a given platform in a single recipe file and use Jinja logic (loops?), rather than splitting builds across a matrix, so that all platform combinations of the `cuda-nvcc-impl_{{ ... }}` package are available during builds of `cuda-nvcc_{{ ... }}`.

Reply:
Also we should skip one of these dependency lines when `target_platform == cross_target_platform`, so that we don't duplicate the dependency for non-cross builds.

Reply:
I'm not totally sure if I follow why we need both `cross_target_platform` and `cross_target_impl_platform`, so maybe I'm missing something here.

Reply:
Yeah essentially this is trying to [mirror this relationship]( https://github.com/conda-forge/ctng-compiler-activation-feedstock/blob/85ab9f32b2868eddda3280956f5abcc8b8c7a021/recipe/meta.yaml#L21-L26 )

When `cross_target_platform != target_platform`, this requires 2 `impl` packages. One for `cross_target_platform` and one for `target_platform`. So those need to be built in advance to satisfy this relationship, which requires `cross_target_impl_platform` to be iterated on separately first to build those `impl` packages

Reply:
Won't that result in builds of the same `impl` package on multiple independent CI jobs? If you already had built `cuda-nvcc-impl_{{ X }}` for every platform, you would only need a simple loop over `cross_target_platform`. But if you loop twice over `cross_target_platform` and `cross_target_impl_platform`, I think you end up building the `impl` packages multiple times.

Consider two separate jobs:
A `linux64` runner wants to build with `cross_target_platform == "linux64"`. It needs to build `cuda-nvcc-impl_linux-64`.
A `linux64` runner wants to build with `cross_target_platform == "aarch64"`. It needs to build `cuda-nvcc-impl_linux-64` and `cuda-nvcc-impl_linux-aarch64`.

You end up with two jobs building `cuda-nvcc-impl_linux-64`, both built with `linux64` platform tags.

Reply:
Yeah was wondering whether we are trying to do too much in this feedstock. Or if this is the right approach for us

In the conda-forge compilers, the `impl` portion is one feedstock and the non-`impl` ones are in another feedstock. This allows checking in the builds from the `impl` step before moving to that next step.

Some questions:

1. Is there a better way to express this relationship?
2. Should we setup a 2nd feedstock to split these up?
3. Should we consolidate packages here (do we need an `impl` or can we fold that in)?

Probably other questions that are worth considering that are not listed above

Reply:
`cross_target_impl_platform` was folded into `cross_target_platform`. So marking this resolved